### PR TITLE
Devtools 104 > Elements > Badges reference

### PR DESCRIPTION
**Pull request by sofiayem**
Documented:
- https://developer.chrome.com/blog/new-in-devtools-104/#slot
- https://developer.chrome.com/blog/new-in-devtools-91/#css-croll-snap
- https://developer.chrome.com/blog/new-in-devtools-94/#container-queries
- https://developer.chrome.com/blog/new-in-devtools-91/#badge-settings
Also documented all other badges.

Publish on Aug 2.

### CODE REVIEW

File: site/en/docs/devtools/elements/badges/index.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+{% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/WMZrfd1AVdMjkgCDqSK8.png", alt="Grid overlay.", width="800", height="674" %}
+
+The overlay shows columns, rows, their numbers, and gaps.
+
+To learn how to debug grid layout, see [Inspect CSS grid](/docs/devtools/css/grid/).
+
+## Flex
+
+An HTML element is a [flexbox](https://developer.mozilla.org/docs/Web/CSS/CSS_Flexible_Box_Layout) if its `display` CSS property is set to `flex` or `inline-flex`. Such elements have `flex` badges next to them that toggle the corresponding overlays.
```
Comment (rachelandrew) [inline site/en/docs/devtools/elements/badges/index.md:52]:
```suggestion
An HTML element is a [flex container](https://developer.mozilla.org/docs/Web/CSS/CSS_Flexible_Box_Layout) if its `display` CSS property is set to `flex` or `inline-flex`. Such elements have `flex` badges next to them that toggle the corresponding overlays.
```

**Review by rachelandrew — APPROVED** [review]
Just a little nit.

File: site/en/docs/devtools/elements/badges/index.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+{% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/MmkJORNp3sOKHHK1CI4h.png", alt="Flex overlay.", width="800", height="646" %}
+
+The overlay shows child element positions. 
+
+To learn how to debug flex layouts, see [Inspect and debug CSS flexbox](/docs/devtools/css/flexbox/).
+
+## Ad
+
+DevTools can detect most ad frames. Such frames have `ad` badges next to them.
```
Comment (jecfish) [inline site/en/docs/devtools/elements/badges/index.md:74]: expand explanation on how we know it's an ad frame. Tag you in a conversation and resource.

File: site/en/docs/devtools/elements/badges/index.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+The [`<slot>` HTML element](https://developer.mozilla.org/docs/Web/HTML/Element/slot) is a placeholder that you can fill with your own content. Together with the `<template>` element,  `<slot>` lets you create separate DOM trees and present them together. Slot content elements have {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/Zml4roc9tWR2DU1Yb9ix.svg", alt="Slot.", width="20", height="20" %}`slot` badges next to them that serve as links to the corresponding slots.
+
+{% Aside %}
+The slot badge is a new DevTools feature available from Chrome version 104. In earlier versions, the preview below doesn't work.
+{% endAside %}
+
+Discover the {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/Zml4roc9tWR2DU1Yb9ix.svg", alt="Slot.", width="20", height="20" %}`slot` badge in the following preview:
+
+{% Codepen {
```
Comment (jecfish) [inline site/en/docs/devtools/elements/badges/index.md:151]: set bigger height, not able to view the code clearly

**Review by jecfish — CHANGES_REQUESTED** [review]

**Review by rachelandrew — APPROVED** [review]
There's a note about a Codepen height to check.

**Comment by sofiayem** [comment]
> There's a note about a Codepen height to check.

Done.

**Pull request merged**